feat(options): Add the option "colorcol" to the fillchars setting to set a character that neovim should draw at the colorcolumn if no other character is there.

jrahm opened the pull request:
Hey all, I've been keeping this little change under my belt for a couple years and thought it might be time to share with the world. All this change does is add an option to the `fillchars` setting: colorcol. This sets the character that neovim should fill the Color Column with when no other character is present. I like this because I find it more sleek to have a 1px line down my color column (by using '│' as my colorcol char) than to have a whole cell highlighted.

Feel free to accept, ignore, whatever, just thought it was time to bring it forth. This is my first neovim pull request, so be gentle please :)

(screenshot is with fillchars+=colorcol:│)

![colorcol](https://user-images.githubusercontent.com/3434593/185770890-52fd59ba-ed2f-49d6-a613-17e8c3375a43.png)

zeertzjq commented on runtime/doc/options.txt:
```diff
@@ -1350,7 +1350,8 @@ A jump table for the options with a short description can be found at |Q_op|.
 	'colorcolumn' is a comma-separated list of screen columns that are
-	highlighted with ColorColumn |hl-ColorColumn|.  Useful to align
+	highlighted with ColorColumn |hl-ColorColumn| and drawn using the
+	colocol option from 'fillchars'.  Useful to align
```
```suggestion
	"colorcol" option from 'fillchars'.  Useful to align
```

lewis6991 commented:
I like this. What do other core members think?

clason commented:
Worth seeing what Bram thinks? Otherwise a small and self-contained change; some bike-shedding on the name maybe, which is a tad long -- `'colcol'`? `'colorc`?

jrahm committed:
feat(colorcolchar): fix typo for colorcolchar documentation

jrahm commented:
I like `colorc` imo

jrahm committed:
feat(colorcolchar): rename colorcol in fillchars to colorc

Rename the colorcol option in fillchars to the more terse colorc.

lewis6991 commented:
FWIW I personally prefer the more verbose version.

jrahm commented:
I'm pretty ambivalent to the name, I'm cool with whatever is considered most idiomatic.

I'm having trouble figuring out why the checks are not successful. Are they flaky?

justinmk commented:
> I'm having trouble figuring out why the checks are not successful. Are they flaky?

No, it's not normal that 5+ CI jobs fail.

> FWIW I personally prefer the more verbose version.

+1 for colorcol.

Though it would be good to shoot this upstream and see if there's interest.

Commit:
feat(colorcolchar): fix highlight_spec.lua

Problem was interpreting a '|' as next command, causing vim to interpret
set fillchars+=colorcol:| incorrectly

Commit:
feat(colorcolchar): revert "feat: rename colorcol in fillchars to colorc"

This reverts commit 234959abbfcf075cb09304b00fc391780580056d and
renames the option 'colorc' -> 'colorcol' again.

jrahm commented:
Fixed the highlight.lua test file and cleaned up the commits.

jrahm committed:
feat(colorcolchar): add the option "colorcol" to the fillchars setting

This option will let neovim draw a character in the colorcolumn when
there is no other character occupying that spot.

For example, I'm someone who likes the elegance of seeing a 1px wide line
at the 80 character mark, rather than a rectangle the width of a cell at
that mark. To accomplish this, I run

:set colorcol=80
:set fillchars=colorcol:│

of course ':' and '.' are good ASCII alteratives.

zeertzjq commented on runtime/doc/options.txt:
```diff
@@ -2478,6 +2479,7 @@ A jump table for the options with a short description can be found at |Q_op|.
 	  diff		'-'		deleted lines of the 'diff' option
 	  msgsep	' '		message separator 'display'
 	  eob		'~'		empty lines at the end of a buffer
+	  colorcol:c	' '		character to display in the colorcolumn
```
```suggestion
	  colorcol	' '		character to display in the colorcolumn
```

zeertzjq commented on runtime/doc/options.txt:
```diff
@@ -2515,6 +2517,7 @@ A jump table for the options with a short description can be found at |Q_op|.
 	  fold		Folded			|hl-Folded|
 	  diff		DiffDelete		|hl-DiffDelete|
 	  eob		EndOfBuffer		|hl-EndOfBuffer|
+	  colorcol:c	ColorColumn		|hl-ColorColumn|
```
```suggestion
	  colorcol	ColorColumn		|hl-ColorColumn|
```

vheon commented:
Since `colorcolumn` is a comma separated list of column to highlight is this going to show that character on each column?

justinmk commented:
> Since `colorcolumn` is a comma separated list of column to highlight is this going to show that character on each column?

see https://github.com/vim/vim/pull/11860
